src/api/ingest/serializers.py

```diff
@@ -0,0 +1,511 @@
+            obs = utils.get_value(record, self.child.observations_field)
+            if key not in groups:
+                utils.set_value(record, self.child.observations_field, value=[obs])
+                groups[key] = record
+            else:
+                utils.get_value(groups[key], self.child.observations_field).append(obs)
+
+        return list(groups.values())
+
+    def create(self, validated_data):
+        records = super().create(validated_data)
+        output = self.group_records(records)
+
+        objs = [
+            CollectRecord(
+                id=rec["id"],
+                stage=rec["stage"],
+                project_id=rec["project"],
+                profile_id=rec["profile"],
+                data=rec["data"],
+            )
+            for rec in output
+        ]
+        return CollectRecord.objects.bulk_create(objs)
+
+    @property
+    def formatted_errors(self):
+        errors = self.errors
+        format_error = self.child.format_error
+
+        fmt_errors = []
+        for error, rec in zip(errors, self._formatted_records):
+            if bool(error) is True:
+                fmt_error = format_error(error)
+                fmt_error["$row_number"] = self._row_index[rec["id"]]
+                fmt_errors.append(fmt_error)
+
+        return sorted(fmt_errors, key=itemgetter("$row_number"))
+
+
+class CollectRecordCSVSerializer(Serializer):
+    protocol = None
+    observations_field = None
+    error_row_offset = 1
+    header_map = {
+        "Site *": "data__sample_event__site",
+        "Management *": "data__sample_event__management",
+        "Sample date: Year *": "data__sample_event__sample_date__year",
+        "Sample date: Month *": "data__sample_event__sample_date__month",
+        "Sample date: Day *": "data__sample_event__sample_date__day",
+        "Sample time": "data__sample_event__sample_time",
+        "Depth *": "data__sample_event__depth",
+        "Interval size": "data__interval_size",
+        "Visibility": "data__sample_event__visibility",
+        "Current": "data__sample_event__current",
+        "Relative depth": "data__sample_event__relative_depth",
+        "Tide": "data__sample_event__tide",
+        "Notes": "data__sample_event__notes",
+        "Observer emails *": "data__observers",
+        "Observation interval *": "data__obs_benthic_pits__interval",
+    }
+
+    # CHOICES
+    _choices = ChoiceViewSet().get_choices()
+    visibility_choices = build_choices("visibilities", _choices)
+    current_choices = build_choices("currents", _choices)
+    relative_depth_choices = build_choices("relativedepths", _choices)
+    tide_choices = build_choices("tides", _choices)
+
+    _reverse_choices = {}
+
+    # PROJECT RELATED CHOICES
+    project_choices = {
+        "data__sample_event__site": None,
+        "data__sample_event__management": None,
+    }
+
+    # FIELDS
+    id = serializers.UUIDField(format="hex_verbose")
```

Comment:
Haven't seen us need to use `hex_verbose` before... Is this a not the default any more?

Reply:
just being explicit, there are 4 options it can be